components/mediation-admin/org.wso2.carbon.localentry/src/main/java/org/wso2/carbon/localentry/service/LocalEntryAdmin.java

```diff
@@ -350,6 +391,28 @@ public OMElement getEntry(String entryKey) throws LocalEntryAdminException {
         return null;
     }
 
+    /**
+     * Returns the OMelement representation of the entry given by sequence
+     * name
+     *
+     * @param entryKey - name of the entry to get
+     * @param tenantDomain - Tenant domain
+     * @return OMElement representing the entryMediator of the given entry
+     *         name
+     * @throws LocalEntryAdminException if any error occured while getting the data from the
+     *                   SynapseConfiguration
+     */
+    public OMElement getEntryForTenant(String entryKey, String tenantDomain) throws LocalEntryAdminException {
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain,
+                    true);
```

Comment:
Shall we split by `PrivilegedCarbonContext.getThreadLocalCarbonContext().` This will make it more readable.

Reply:
Updated